flytekit/clients/auth_helper.py

```diff
@@ -34,12 +36,14 @@ def get_client_config(self) -> ClientConfig:
         metadata_service = AuthMetadataServiceStub(self._secure_channel)
         public_client_config = metadata_service.GetPublicClientConfig(PublicClientAuthConfigRequest())
         oauth2_metadata = metadata_service.GetOAuth2Metadata(OAuth2MetadataRequest())
+        # Use the ones defined locally (if any) or default to `public_client_config.scopes`
+        scopes = self._scopes or public_client_config.scopes
         return ClientConfig(
             token_endpoint=oauth2_metadata.token_endpoint,
             authorization_endpoint=oauth2_metadata.authorization_endpoint,
             redirect_uri=public_client_config.redirect_uri,
             client_id=public_client_config.client_id,
-            scopes=public_client_config.scopes,
```

Comment:
i think we should not mess around with this, but add scopes to the constructor of individual authenticators. this way it remains clear

Reply:
mmm, okay. Would you add it for all authenticators? Or just for `ClientCredentialsAuthenticator`?

Reply:
let's just do the clientcredentials one for now please.

Reply:
Cool. Done!